staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/managedfields.go

```diff
@@ -110,7 +110,30 @@ func EncodeObjectManagedFields(obj runtime.Object, managed ManagedInterface) err
 func decodeManagedFields(encodedManagedFields []metav1.ManagedFieldsEntry) (managed managedStruct, err error) {
 	managed.fields = make(fieldpath.ManagedFields, len(encodedManagedFields))
 	managed.times = make(map[string]*metav1.Time, len(encodedManagedFields))
-	for _, encodedVersionedSet := range encodedManagedFields {
+
+	// If we have exactly one managedFields with no manager and no
+	// version, keep an empty manager to get a reset.
+	if len(encodedManagedFields) == 1 {
+		if mf := encodedManagedFields[0]; mf.Manager == "" && mf.APIVersion == "" {
+			fields, err := decodeVersionedSet(&mf)
+			if err != nil {
+				return managedStruct{}, err
+			}
+			managed.fields["resetManager"] = fields
+			managed.times["resetManager"] = nil
```

Comment:
is "resetManager" a magic string? Does it drive behavior elsewhere?

Reply:
This is more subtle than I wish this was. This is just an empty manager with a random name that can't conflict (managers names internally are actually json objects).

Empty managers are going to be treated as a drop, and will be replaced with the managedFields from the live object. The best way to reset the fieldManager is to insert an empty `ManagedFieldsEntry` so that the list it's not empty, but the dummy entry will be dropped on serialization.